src/Microsoft.AspNet.OData.Shared/Query/DefaultQuerySettings.cs

```diff
@@ -64,5 +64,16 @@ public class DefaultQuerySettings
         /// Gets or sets a value indicating whether the service will use skiptoken or not.
         /// </summary>
         public bool EnableSkipToken { get; set; }
+
+        internal void CopySettings(DefaultQuerySettings settings)
+        {
+            this.EnableCount = settings.EnableCount;
+            this.EnableExpand = settings.EnableExpand;
+            this.EnableFilter = settings.EnableFilter;
+            this.EnableOrderBy = settings.EnableOrderBy;
+            this.EnableSelect = settings.EnableSelect;
+            this.EnableSkipToken = settings.EnableSkipToken;
+            this.MaxTop = settings.MaxTop;
```

Comment:
Should we only copy this over if this.MaxTop.HasValue == true, or would we need to copy it over in order to allow clearing a MaxTop through DI?

Reply:
I think we need to copy to reset the MaxTop to even null.